lib/collection.js

```diff
@@ -8,22 +8,43 @@ const Promise = require('bluebird');
 const createError = require('create-error');
 
 /**
+ * When creating a {@link Collection}, you may choose to pass in the initial array of
+ * {@link Model models}. The collection's {@link Collection#comparator comparator} may be included
+ * as an option. Passing `false` as the comparator option will prevent sorting. If you define an
+ * {@link Collection#initialize initialize} function, it will be invoked when the collection is
+ * created.
+ *
+ * If you would like to customize the Collection used by your models when calling
+ * {@link Model#fetchAll} or {@link Model#fetchPage} you can use the following process:
+ *
+ *     const Test = bookshelf.model('Test', {
+ *       tableName: 'test'
+ *     }, {
+ *       collection(...args) {
+ *         return new Tests(...args)
+ *       }
+ *     })
+ *     const Tests = bookshelf.collection('Tests', {
+ *       get model() {
+ *         return Test
+ *       },
+ *       doStuff() {
+ *         // This method will be available in the results collection returned
+ *         // by Test.fetchAll() and Test.fetchPage()
+ *       }
+ *     })
```

Comment:
This looks good, but I would add that fetching event also works, e.g.

```
  *     const Tests = bookshelf.collection('Tests', {
  *       get model() {
  *         return Test
  *       },
  *.      initialize (...args) {
  *         super.initialize(...args)
  *         // Collection will emit fetching event as expected even on eager queries. 
  *.        this.on('fetching', () => {})
  *       },
  *       doStuff() {
  *         // This method will be available in the results collection returned
  *         // by Test.fetchAll() and Test.fetchPage()
  *       }
  *     })
```

Reply:
That is indeed a very useful addition to the example. I'll add it later today.

Reply:
Done.